### PR TITLE
Added Record toEntries, fromEntries

### PR DESCRIPTION
**Pull request by bravely**
* Added `Record.toEntries` and `Record.fromEntries`, to facilitate a common need of folks coming over from 'normal' JS.

### CODE REVIEW

**Comment by mlegenhausen** [comment]
`toEntries` is `toReadonlyArray` https://github.com/gcanti/fp-ts/blob/master/src/ReadonlyRecord.ts#L136-L138

`fromEntries` is `fromFoldable`

```ts
import * as RR from 'fp-ts/ReadonlyRecord'
import * as RA from 'fp-ts/ReadonlyArray'
import * as S from 'fp-ts/Semigroup'

RR.fromFoldable(S.last<string>(), RA.Foldable)([['a', 'b'], ['c', 'd'], ['a', 'e']])
// { a: 'e', c: 'd' }
```

**Comment by bravely** [comment]
@mlegenhausen Didn't realize `toEntries` was `toArray`/`toReadonlyArray`, that works.

The other though, I feel like there should be a convenience for, especially considering we could easily auto-type it.

**Comment by mlegenhausen** [comment]
@bravely sorry overlooked that you just defined an alias for `fromFoldable`.

If you could adjust your PR so the `toEntries` and `fromEntries` are defined in `ReadonlyRecord` and just reexported in `Record` the last decision lies by @gcanti.

IMHO it would be a good addition especially for people who are new `fp-ts` programmers and for me  `RR.fromFoldable(S.last<A>(), RA.Foldable)` is the most used configuration.

**Comment by bravely** [comment]
@mlegenhausen I appreciate the support! I matched the feature in `ReadonlyRecord`, however I couldn't simply alias the result because of the nature of the `readonly` parameters. So I made them effectively match, with `ReadonlyRecord` creating a `ReadonlyArray` and `Record` creating an `Array`.

**Comment by mlegenhausen** [comment]
Looks good for me could you please add `@since 2.12.0` to your comments?

**Comment by bravely** [comment]
@mlegenhausen Apologies for how long it took, but I've added those now.

**Comment by mlegenhausen** [comment]
@bravely please check the CI result. In this case you need to run `npm run fix-prettier`.

**Comment by bravely** [comment]
@mlegenhausen Should be good to go now!

**Review by mlegenhausen — APPROVED** [review]

**Comment by steida** [comment]
I just realized I need such helpers.  @gcanti Can you please merge it? Thank you.

File: src/ReadonlyRecord.ts [hunk 1]
```diff
@@ -29,6 +30,8 @@ import { TraversableWithIndex1 } from './TraversableWithIndex'
 import { Unfoldable, Unfoldable1 } from './Unfoldable'
 import { PipeableWilt1, PipeableWither1, wiltDefault, Witherable1, witherDefault } from './Witherable'
 
+type Semigroup<A> = Se.Semigroup<A>
```
Comment (gcanti) [inline src/ReadonlyRecord.ts:33]: @bravely I believe this can be replaced by

```ts
import Semigroup = Se.Semigroup
```

File: src/Record.ts [hunk 2]
```diff
@@ -30,6 +31,8 @@ import { TraversableWithIndex1 } from './TraversableWithIndex'
 import { Unfoldable, Unfoldable1 } from './Unfoldable'
 import { PipeableWilt1, PipeableWither1, wiltDefault, Witherable1, witherDefault } from './Witherable'
 
+type Semigroup<A> = Se.Semigroup<A>
```
Comment (gcanti) [inline src/Record.ts:34]: idem

**Review by gcanti — CHANGES_REQUESTED** [review]

**Comment by bravely** [comment]
@gcanti Updated!

**Review by gcanti — APPROVED** [review]

**Comment by gcanti** [comment]
Thank you @bravely 👍

**Pull request merged**